linera-views/src/backends/scylla_db.rs

```diff
@@ -673,23 +713,28 @@ impl AdminKeyValueStore for ScyllaDbStoreInternal {
                     }
                 }
             };
-            if let Some(rows) = result.rows {
-                // The output of the description is the following:
-                // * The first column is the keyspace (in that case kv)
-                // * The second column is the nature of the object.
-                // * The third column is the name
-                // * The fourth column is the command that built it.
-                for row in rows.into_typed::<(String, String, String, String)>() {
-                    let (_, object_kind, name, _) = row?;
-                    if object_kind == "table" {
-                        namespaces.push(name);
-                    }
+
+            for row in result
+                .into_rows_result()?
+                .rows::<(String, String, String, String)>()?
+            {
+                let (_, object_kind, name, _) = row?;
+                if object_kind == "table" {
+                    namespaces.push(name);
                 }
             }
-            if result.paging_state.is_none() {
-                return Ok(namespaces);
+
+            match paging_state_response.into_paging_control_flow() {
+                ControlFlow::Break(()) => {
+                    // No more pages to be fetched.
+                    return Ok(namespaces);
+                }
+                ControlFlow::Continue(new_paging_state) => {
+                    // Update paging state from the response, so that query
+                    // will be resumed from where it ended the last time.
+                    paging_state = new_paging_state;
+                }
```

Comment:
Isn't this just a manual implementation of `query_iter`?

Reply:
Correct, I removed that now.